students/smitco/lesson09/mailroomOO.py

```diff
@@ -5,65 +5,47 @@
 import os
 import datetime
 
-donors = {"John Travolta": [5000, 7500],
-          "Jane Fonda": [10000, 8000, 6500],
-          "Judy Blume": [3000, 3000],
-          "Joey Tribbiani": [9000],
-          "Jenny Gump": [10300, 13750, 12500]}
 
 class Donor(object):
```

Comment:
your Donor class doesn't seem to serve a lot of purpose right now.

Donor class should hold everything about the single Donor and be integrated with AllDonors; here is some example code:
``` 
class Donor:
    def __init__(self, name, donations):
        self.name = name  # ONLY donor name, no "list"
        self.donations = donations

    @property
    def total(self):
        return sum(self.donations)  # example

    @property
    def average(self):
        pass

    @property
    def num_donations(self):
        pass

    def add(self, donation):
        self.donations.append(donation)

    def get_letter_text(self):
        return f"Letter stuff for {self.name}"


class Donors:
    def __init__(self, donors=None):
        self.donors = donors or {}  # initialize
        # the data structure here is a dictionary with donor name as a key and donor object reference as a value

    @classmethod
    def initialize(cls, *donor_objs):  # getting really fancy here
        return cls(donors={d.name: d for d in donor_objs})

    def add_donation(self, donor_name, donation):
        if donor_name in self.donors:
            self.donors[donor_name].add(donation)
        
        else:
            self.donors[donor_name] = Donor(donor_name, [donation])
    
    def get_report(self):
        """Return report as a list of rows (could also return a string)"""
        rows = []
        for donor in self.donors.values():
            rows.append(f"{donor.name} {donor.total}")
        return rows
    
    def send_letters(self):
        for donor in self.donors.values():
            print(donor.get_letter_text())

    
```
example usage:
```
donors = Donors.initialize(Donor("Natasha", [100]), Donor("Tom", [999]))
print(donors.get_report())
```